akanda/nova/neutronv2/api.py

```diff
@@ -275,32 +309,30 @@ def deallocate_for_instance(self, context, instance, **kwargs):
                 LOG.info(_('Unable to reset device ID for port %s'), port,
                          instance=instance)
 
+        # ---------------------------------------------------------------------
+        # NOTE(rods): check the value of the device_owner
+        #
+        # original code
+        # self._delete_ports(neutron, instance, ports, raise_if_fail=True)
+
+        ports_to_delete = []
         for port in ports:
-            try:
-                # -------------------------------------------------------------
-                # NOTE(rods): check the value of the device_owner
-                device_owner = port['device_owner']
-                port = port['id']
-                if device_owner.startswith('network:'):
-                    body = dict(device_id='')
-                    neutron.update_port(port, dict(port=body))
-                    # ---------------------------------------------------------
-                else:
-                    neutron.delete_port(port)
-            except neutronv2.exceptions.NeutronClientException as e:
-                if e.status_code == 404:
-                    LOG.warning(_("Port %s does not exist"), port)
-                else:
-                    with excutils.save_and_reraise_exception():
-                        LOG.exception(_("Failed to delete neutron port %s"),
-                                      port)
+            if port['device_owner'].startswith('network:'):
+                body = dict(device_id='')
+                neutron.update_port(port['id'], dict(port=body))
+            else:
+                ports_to_delete.append(port['id'])
+
+        self._delete_ports(neutron, instance, ports_to_delete,
```

Comment:
We're no longer doing any exception handling here; can you give any explanation (e.g., the exception handling moved into `_delete_ports`)?